### PR TITLE
add configurable tiller probe port

### PR DESCRIPTION
**Pull request by Ewocker**

### CODE REVIEW

File: cmd/helm/init.go [hunk 1]
```diff
@@ -151,6 +151,8 @@ func newInitCmd(out io.Writer) *cobra.Command {
 	f.StringArrayVar(&i.opts.Values, "override", []string{}, "override values for the Tiller Deployment manifest (can specify multiple or separate values with commas: key1=val1,key2=val2)")
 	f.BoolVar(&i.opts.AutoMountServiceAccountToken, "automount-service-account-token", true, "auto-mount the given service account to tiller")
 
+	f.IntVar(&i.opts.ProbeServerPort, "probe-server-port", 0, "overrides the default probe server port for Tiller")
```
Comment (johngmyers) [inline cmd/helm/init.go]: Description "probe server port for Tiller"

Default value is 44135, not 0

File: cmd/tiller/tiller.go [hunk 2]
```diff
@@ -65,14 +65,15 @@ const (
...
-	probeAddr = ":44135"
 	traceAddr = ":44136"
 
 	// defaultMaxHistory sets the maximum number of releases to 0: unlimited
 	defaultMaxHistory = 0
 )
 
 var (
+	probeAddr            = flag.String("probe-address", ":44135", "address:port to listen on for probe server")
+	probeEnable          = flag.Bool("probe", true, "enable probe server")
```
Comment (johngmyers) [inline cmd/tiller/tiller.go:76]: Being able to disable probe server is a separate concern and should be in a separate commit. Do we even need the option?

**Review by johngmyers — CHANGES_REQUESTED** [review]

File: cmd/tiller/tiller.go (hunk 2 above)
Comment (Ewocker) [reply cmd/tiller/tiller.go:76]: I am currently using this option. Using local tiller does not need to enable liveness and readiness probe.

Comment (Ewocker) [reply cmd/tiller/tiller.go:76]: So technically the one that can be removed is probeAddr. Reason for having those was because the code change is based on this PR https://github.com/helm/helm/pull/3440

Comment (johngmyers) [reply cmd/tiller/tiller.go:76]: They should still be separate commits. Only the enable flag commit would be fine.

Comment (Ewocker) [reply cmd/tiller/tiller.go:76]: sure, I will remove the configuration part.

**Review by johngmyers — APPROVED** [review]
